examples/smartcore-ml/README.md

```diff
@@ -24,65 +24,67 @@ when using KNN, be sure to add the following import:
 
-```
+```rust
 let model: DecisionTreeClassifier<{float}, u32, DenseMatrix<{float}>, Vec<u32>> = trained_model;
 ```
 
-```
+```rust
 let model: RandomForestClassifier<{float}, u8, DenseMatrix<{float}>, Vec<u8>> = trained_model;
 ```
 
-```
+```rust
 let model: GaussianNB<{float}, _, DenseMatrix<{float}>, _> = trained_model;
 ```
 
-```
+```rust
 let model: LogisticRegression<{float}, u32, DenseMatrix<{float}>, Vec<u32>> = trained_model;
 ```
 
-```
+```rust
 let model: LinearRegression< {float}, {float}, DenseMatrix<{float}>, Vec<{float}>> = trained_model;
 ```
 
-```
+```rust
 let model: KNNClassifier< {float}, u32, DenseMatrix<{float}>, Vec<u32>, Euclidian<{float}>> = trained_model;
 ```
 
-```
+```rust
 let model: KMeans<{float}, u8, DenseMatrix<{float}>, Vec<u8>> = trained_model;
 ```
 
-```
+```rust
 let model: PCA<{float}, DenseMatrix<{float}>> = trained_model;
 ```
 
 NOTE:  Deserialization of SVC and SVR does not include that `SVCParametersmake` field.  It must be added back into the model struct manually after deserialization.  The RISC Zero SmartCore fork has amended the visibility of the model `struct`, making the parameters field public and thereby allowing the parameters field to be directly inserted into the model struct after deserialization.  You must insert the same parameters that were used when training the model.
-```
+
+```rust
 let mut model: SVC<{float}, i32, DenseMatrix<{float}>, Vec<i32>> = deserialized_svc_trained;
 let params = &SVCParameters::default().with_c(200.0).with_kernel(Kernels::linear());
 model.parameters = Some(params);
 ```
 
-```
+```rust
 let mut model: SVR< '_, {float}, DenseMatrix<{float}>, Vec<{float}>> = deserialized_model;
 //make sure to use the same SVRParameters used to train the model
 let params = &SVRParameters::default().with_eps(2.0).with_c(10.0).with_kernel(Kernels::linear());
 model.parameters = params;
 ```
 
 SVM models support four kernels:  Gaussian (rbf), Linear, Polynomial, and Sigmoid.  Kernels are selected within the `.with_kernel()` function in the `SVCParameters` or `SVRParameters`.  Specify the desired kernel as follows:
-* Kernel::linear()
-* Kernel::rbf()
-* Kernel::polynomial()
-* Kernel::sigmoid()
+
+- Kernel::linear()
+- Kernel::rbf()
+- Kernel::polynomial()
+- Kernel::sigmoid()
 
 ## SmartCore
 
-The source code for SmartCore can be found here:  https://github.com/smartcorelib/smartcore
+The source code for SmartCore can be found here:  [https://github.com/smartcorelib/smartcore](https://github.com/smartcorelib/smartcore)
 
-You can find other model implementations and additional SmartCore features here:  https://github.com/smartcorelib/smartcore-jupyter/tree/main/notebooks
+You can find other model implementations and additional SmartCore features here:  [https://github.com/smartcorelib/smartcore-jupyter/tree/main/notebooks](https://github.com/smartcorelib/smartcore-jupyter/tree/main/notebooks)
```

Comment:
hmm... not clear to me this is an improvement, but i'm fine with it.

Reply:
i guess relatedly... mb we want to move links to the bottom of the page in example READMEs also?

Reply:
oh interesting -- in the `waldo` example it looks like links *are* being moved to the bottom of the doc. 

maybe this is a situation where the first linting converts from naked url to []() syntax and the second linting would move the link to the bottom of the page?

Reply:
so the thing with naked URLs is that normally you'd do something like `<https://github.com/smartcorelib/smartcore>` but, in the case of docusaurus (I think related to MDX) using `<https://github.com/smartcorelib/smartcore>` breaks the page build. So I had to put a check to enforce the `[<https://github.com/smartcorelib/smartcore>](<https://github.com/smartcorelib/smartcore>)` format, which looks redundant but is the only way to make it work in docusaurus

Reply:
if we want, we could leverage https://github.com/remarkjs/remark-reference-links or remark-defsplit which would turn all links in reference links, thus making it more consistent. But.... ehh I'm not sure

Reply:
you could also do the opposite automatically with remark, example
```
# Local Proving

RISC Zero offers a fully [open-source] prover.
Users can run the [zkVM] locally, generating proofs using their own hardware.
The [feature flags] include options to toggle between CPU and GPU proving.

## Local vs. Remote Proving

For most use cases, we recommend remote proving using [Bonsai].
The primary reason to consider using local proof generation for your application is to manage private data.

> Whoever is generating the proofs can see all private information involved.

_With local proof generation, you can generate proofs about your private data, while your private data never leaves your machine._

[Bonsai]: ./remote-proving.md
[feature flags]: https://github.com/risc0/risc0#feature-flags
[open-source]: https://risczero.com/news/open-source
[zkVM]: ../zkvm/zkvm-overview.md
```

would become
```
# Local Proving

RISC Zero offers a fully [open-source](https://risczero.com/news/open-source) prover.
Users can run the [zkVM](../zkvm/zkvm-overview.md) locally, generating proofs using their own hardware.
The [feature flags](https://github.com/risc0/risc0#feature-flags) include options to toggle between CPU and GPU proving.

## Local vs. Remote Proving

For most use cases, we recommend remote proving using [Bonsai](./remote-proving.md).
The primary reason to consider using local proof generation for your application is to manage private data.

> Whoever is generating the proofs can see all private information involved.

_With local proof generation, you can generate proofs about your private data, while your private data never leaves your machine._
```

Reply:
@nuke-web3 do you have an opinion on this one?

Reply:
On my end, strong preference to this version, both for readability and for maintenance

```
# Local Proving

RISC Zero offers a fully [open-source] prover.
Users can run the [zkVM] locally, generating proofs using their own hardware.
The [feature flags] include options to toggle between CPU and GPU proving.

## Local vs. Remote Proving

For most use cases, we recommend remote proving using [Bonsai].
The primary reason to consider using local proof generation for your application is to manage private data.

> Whoever is generating the proofs can see all private information involved.

_With local proof generation, you can generate proofs about your private data, while your private data never leaves your machine._

[Bonsai]: ./remote-proving.md
[feature flags]: https://github.com/risc0/risc0#feature-flags
[open-source]: https://risczero.com/news/open-source
[zkVM]: ../zkvm/zkvm-overview.md